src/main/java/core/basesyntax/service/impl/DataConverterImpl.java

```diff
@@ -0,0 +1,22 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.service.DataConverter;
+import java.util.List;
+
+public class DataConverterImpl implements DataConverter {
+    private static final int INDEX_OPERATION = 0;
+    private static final int INDEX_FRUIT = 1;
+    private static final int INDEX_QUANTITY = 2;
+
+    @Override
+    public List<FruitTransaction> convertToTransaction(List<String> inputReport) {
+        return inputReport.stream()
+                .map(string -> string.split(","))
+                .map(strings -> new FruitTransaction(
+                        strings[INDEX_OPERATION],
+                        strings[INDEX_FRUIT],
+                        strings[INDEX_QUANTITY]))
+                .toList();
```

Comment:
The method 'convertToTransaction' lacks error handling. If any of the input strings are malformed (e.g., missing elements or incorrect data types), this will result in an ArrayIndexOutOfBoundsException or other runtime exceptions. Consider adding validation for the input strings and appropriate error handling to ensure that each string can be successfully converted into a FruitTransaction object.

Reply:
++